modules/040-node-manager/hooks/machine_deployments_caps_metrics.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook/metrics"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
+
+	"github.com/deckhouse/deckhouse/modules/040-node-manager/hooks/internal/capi/v1beta1"
+)
+
+const (
+	capsMachineDeploymentMetricsGroup          = "caps_md"
+	capsMachineDeploymentMetricReplicasName    = "d8_caps_md_replicas"
+	capsMachineDeploymentMetricDesiredName     = "d8_caps_md_desired"
+	capsMachineDeploymentMetricReadyName       = "d8_caps_md_ready"
+	capsMachineDeploymentMetricUnavailableName = "d8_caps_md_unavailable"
+	capsMachineDeploymentMetricPhaseName       = "d8_caps_md_phase"
+)
+
+type machineDeploymentStatus struct {
+	Name        string
+	Replicas    float64
+	Desired     float64
+	Ready       float64
+	Unavailable float64
+	Phase       float64
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: "/modules/node-manager/metrics",
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:                   "machinedeployment_status",
+			ApiVersion:             "cluster.x-k8s.io/v1beta1",
+			Kind:                   "MachineDeployment",
+			WaitForSynchronization: pointer.Bool(false),
```

Comment:
We can have hybrid cluster (for example zVirt with static instances) and we will have alert not only with CAPS instances and zVirt machine deployments.

Better way: add additional label here https://github.com/deckhouse/deckhouse/blob/eb1bc03155dd862149b2f8e93d39a6f7e3464461/modules/040-node-manager/templates/node-group/_capi_machine_deployment.tpl#L21-L20
and filter by this label

Reply:
I just clarified one thing and it looks like adding a label will result in a machine deployment rollout https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine-deployment#in-place-propagation
Sorry for the wrong tip. Let's add an annotation, subscribe to all MachineDeployments and filter by annotation.

Reply:
I've checked that adding a label in the machinedeployment doesn't trigger a rollout in the machineset/machine or nodes. Perhaps this documentation page is incomplete.